clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py

```diff
@@ -61,18 +65,13 @@ def run(self, progress_callback):
         # Build deduplicated lists of input/output datasets:
         input_datasets = []
         output_datasets = []
-        for rcp in all_recipes:
-            rcp_inputs_dict = rcp["inputs"]
-            rcp_outputs_dict = rcp["outputs"]
+        for recipe in all_recipes:
+            recipe_inputs_dict = recipe["inputs"]
+            recipe_outputs_dict = recipe["outputs"]
             # CASE: no input dataset
-            if rcp_inputs_dict:
-                input_key = list(rcp_inputs_dict.keys())[0]
-                rcp_inputs_list = [x["ref"] for x in rcp_inputs_dict[input_key]["items"]]
-                input_datasets += rcp_inputs_list
-            output_key = list(rcp_outputs_dict.keys())[0]
-            rcp_outputs_list = [x["ref"] for x in rcp_outputs_dict[output_key]["items"]]
-            # Append them to the overall input list:
-            output_datasets += rcp_outputs_list
+            if recipe_inputs_dict:
+                append_datasets_in_dict_to_list(recipe_inputs_dict, input_datasets)
+            append_datasets_in_dict_to_list(recipe_outputs_dict, output_datasets)
         # Deduplicate input/output lists:
         input_datasets = list(set(input_datasets))
         output_datasets = list(set(output_datasets))
```

Comment:
If a same dataset is used in a lot of recipes in a flow, it is not optimal to add the dataset name those number of time and then deduplicate the list. Let's remove those lines and add a condition to not add a dataset already in the list in  `append_datasets_in_dict_to_list`

Reply:
Lines removed in commit [9e8d7871ee0bb75589c383edb517a7bd702f070e](https://github.com/gbetegon88/dataiku-contrib/commit/9e8d7871ee0bb75589c383edb517a7bd702f070e)